src/docfx/lib/watch/Watch{T}.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+
+namespace Microsoft.Docs.Build
+{
+    public class Watch<T>
+    {
+        private readonly Func<T> _valueFactory;
+
+        [MaybeNull]
+        private T _value;
+
+        private volatile ContainerFunction? _function;
+        private object? _syncLock;
+
+        public Watch(Func<T> valueFactory) => _valueFactory = valueFactory;
+
+        public bool IsValueCreated => _function != null;
```

Comment:
>IsValueCreated [](start = 20, length = 14)

will this just be used for testing?

Reply:
Yes for now. It exist because `Lazy<T>` also has this public property. I want `Watch<T>` to be a replacement of `Lazy<T>`.